stores/pg/src/vector/index.ts

```diff
@@ -215,18 +216,52 @@ export class PgVector extends MastraVector {
         throw new Error('PostgreSQL vector extension is not available. Please install it first.');
       }
 
-      // Try to create extension
-      await client.query('CREATE EXTENSION IF NOT EXISTS vector');
+      // Get advisory lock using hash of index name
+      const hash = createHash('sha256').update(indexName).digest('hex');
+      const lockId = BigInt('0x' + hash.slice(0, 8)) % BigInt(2 ** 31); // Take first 8 chars and convert to number
+      const acquired = await client.query('SELECT pg_try_advisory_lock($1)', [lockId]);
+
+      if (!acquired.rows[0].pg_try_advisory_lock) {
+        // Check if table already exists
+        const exists = await client.query(
+          `
+          SELECT 1 FROM pg_class c 
+          JOIN pg_namespace n ON n.oid = c.relnamespace
+          WHERE c.relname = $1 
+          AND n.nspname = 'public'
+        `,
+          [indexName],
+        );
+
+        if (exists.rows.length > 0) {
+          // Table exists, just build index if needed
+          if (buildIndex) {
+            await this.buildIndex({ indexName, metric, indexConfig });
```

Comment:
wont this run into the same issue as before since we're building the index without a lock?

Reply:
I have a lock inside buildIndex too